packages/material-ui/src/Paper/Paper.js

```diff
@@ -91,7 +86,17 @@ Paper.propTypes = {
    * Shadow depth, corresponds to `dp` in the spec.
    * It accepts values between 0 and 24 inclusive.
    */
-  elevation: PropTypes.number,
+  elevation: chainPropTypes(PropTypes.number, (props) => {
+    const { classes, elevation } = props;
+    // in case `withStyles` fails to inject we don't need this warning
+    if (classes === undefined) {
+      return null;
+    }
+    if (elevation != null && classes[`elevation${elevation}`] === undefined) {
```

Comment:
Should we consider #20064 in the equation?

Reply:
That issue should consider the new implementation, yes.

Reply:
Ok, I will update the diff proposal with a new fix once this one is merged.